numba_dpcomp/numba_dpcomp/mlir_compiler/lib/pipelines/lower_to_llvm.cpp

```diff
@@ -1755,6 +1755,44 @@ struct LowerReleaseContextOp
   }
 };
 
+// Convert operations from the plier_util dialect to the LLVM dialect.
+struct PierUtilToLLVMPass
+    : public mlir::PassWrapper<PierUtilToLLVMPass,
+                               mlir::OperationPass<mlir::ModuleOp>> {
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(PierUtilToLLVMPass)
+
+  PierUtilToLLVMPass() = default;
+
+  void runOnOperation() override {
+    mlir::Operation *op = getOperation();
+    auto &context = getContext();
+    auto options = getLLVMOptions(context);
+
+    mlir::LLVMTypeConverter typeConverter(&context, options);
+    populateToLLVMAdditionalTypeConversion(typeConverter);
+    mlir::RewritePatternSet patterns(&context);
+
+    patterns.insert<
+        // clang-format off
+        LowerUndef,
+        LowerBuildTuple,
+        LowerRetainOp,
+        ReshapeLowering,
+        ExpandShapeLowering,
+        LowerExtractMemrefMetadataOp,
+        LowerTakeContextOp,
+        LowerReleaseContextOp
+        // clang-format on
+        >(typeConverter);
+
+    mlir::LLVMConversionTarget target(context);
+    target.addLegalOp<mlir::func::FuncOp>();
+    target.addIllegalOp<plier::TakeContextOp, plier::ReleaseContextOp>();
```

Comment:
If this is lowering plier_util to llvm, why doesn't it declare plier_util as illegal?

Reply:
This was mostly covered by input/output types legality check already present `LLVMTypeConverter`, but it won't hurt I guess.

Reply:
done